src/components/FilterList.js

```diff
@@ -1,66 +1,53 @@
 import './ShowList.css';
 import ShowList from './ShowList';
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from "react-redux";
 
 
-function FilterList({ taskList, setTaskList , filter }){
 
-const pending = filter.pending;
-const inProgress = filter.inProgress;
-const done = filter.done;
-    
-
-   
-
-       return(
-           <div>
-            {
-                taskList.map((task) => {
-                    if (pending === true && inProgress === true && done === true){
-                        if (task.currentStatus === "Pending" || "In Progress" || "Done"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
-
-                    else if (pending === true && inProgress === true && done === false){
-                        if (task.currentStatus !== "Done"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
-
-                    else if (pending === true && inProgress === false && done === true){
-                        if (task.currentStatus !== "In Progress"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
-
-                    else if (pending === false && inProgress === true && done === true){
-                        if (task.currentStatus !== "Pending"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
+function FilterList({ filter }){
 
-                    else if (pending === true && inProgress === false && done === false){
-                        if (task.currentStatus === "Pending"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
+const taskList = useSelector( state => state.taskList );
+const updatedList = useSelector( state => state.updatedList );
+let List;
+const [tasks, setTasks] = useState([]);
+const dispatch = useDispatch();
 
-                    else if (pending === false && inProgress === true && done === false){
-                        if (task.currentStatus === "In Progress"){
-                            return(<ShowList task={task} taskList={taskList} setTaskList={setTaskList}/>);
-                    }}
+useEffect(() => {
+    
+        List = ((taskList.filter((task) => {
+            if (task.status === "Pending" && filter.pending === true){
+                return task;
+            }
+            if (task.status === "In Progress" && filter.inProgress === true){
+                return task;
+            }
+            if (task.status === "Done" && filter.done === true){
+                return task;
+            }
+            
+        }))) 
+        
+        
+        setTasks(List);
+        dispatch({
```

Comment:
why did you dispatch tasks ? It's a component state just to store filter state

Reply:
I stored the filterd tasks on redux state and used it to display the list

Reply:
do we really need it ? Since we can just pass those filter lists as props to other components